ros/src/computing/perception/detection/packages/lidar_tracker/nodes/obj_fusion/obj_fusion.cpp

```diff
@@ -123,151 +151,203 @@ static void fusion_objects(void) {
+            qm_lane.setRPY(0, 0, atan2(fp.bx - bp.bx, fp.ly - bp.ly)); // y,x
+            tf::Quaternion q_step;  // 90 deg
+            q_step.setRPY(0, 0, M_PI/2);
+            double r_max = M_PI;
+            // search in 0,90,180,270-degree
+            tf::Quaternion qr_obj = q_obj; // rotated
+            for (int i = 0; i < 4; i++) {
+              tf::Quaternion qrm_obj = tform * qr_obj;  // map-cood
+              double r = qm_lane.angle(qrm_obj);
+              r = (r >= M_PI/2) ? (r - M_PI):r;
+              if (fabs(r) < r_max) {
+                r_max = fabs(r);
+                rot_n = i;
+              }
+              qr_obj *= q_step;
+            }
+            double roll, pitch, yaw;
+            tf::Matrix3x3(q_obj).getRPY(roll, pitch, yaw);
+            ROS_INFO(" %d roll=%f pitch=%f yaw=%f", rot_n*90, roll, pitch, yaw);
+            break;
+          }
+        }
+      } else {
+        ROS_INFO("%s: VectorMap Server call failed.", __FUNCTION__);
+      }
+      // determine rotation
+      tf::Quaternion dq_obj;
+      dq_obj.setRPY(0, 0, M_PI*rot_n/2);
+      q_obj *= dq_obj;
+      // bounding_box
+      bounding_box.pose.orientation.x = q_obj.x();
+      bounding_box.pose.orientation.y = q_obj.y();
+      bounding_box.pose.orientation.z = q_obj.z();
+      bounding_box.pose.orientation.w = q_obj.w();
+      if (rot_n % 2 == 1) { // swap x-y at 90,270 deg
+        std::swap(bounding_box.dimensions.x, bounding_box.dimensions.y);
+      }
+      // cloud clusters
+      v_cloud_cluster.at(obj_indices.at(i)).bounding_box = bounding_box;
+    }
+
+    // x-axis
+    visualization_msgs::Marker marker;
+    marker.header = header;
+    marker.id = marker_id++;
+    marker.lifetime = ros::Duration(0.1);
+    marker.type = visualization_msgs::Marker::ARROW;
+    marker.pose.position = bounding_box.pose.position;
+    marker.pose.orientation = bounding_box.pose.orientation;
+    marker.scale.x = 2.0;
+    marker.scale.y = 0.2;
+    marker.scale.z = 0.1;
+    marker.color.r = 1.0;
+    marker.color.a = 1.0;
+    marker_array_msg.markers.push_back(marker);
+
+    // y-axis
+    tf::Quaternion qy;
+    qy.setRPY(0, 0, M_PI/2);
+    q_obj *= qy;
+    marker.id = marker_id++;
+    marker.pose.orientation.x = q_obj.x();
+    marker.pose.orientation.y = q_obj.y();
+    marker.pose.orientation.z = q_obj.z();
+    marker.pose.orientation.w = q_obj.w();
+    marker.color.r = 0.0;
+    marker.color.g = 1.0;
+    marker.color.a = 1.0;
+    marker_array_msg.markers.push_back(marker);
+
+    // z-axis
+    tf::Quaternion qz;
+    qz.setRPY(0, -M_PI/2, 0);
+    q_obj *= qz;
+    marker.id = marker_id++;
+    marker.pose.orientation.x = q_obj.x();
+    marker.pose.orientation.y = q_obj.y();
+    marker.pose.orientation.z = q_obj.z();
+    marker.pose.orientation.w = q_obj.w();
+    marker.color.g = 0.0;
+    marker.color.b = 1.0;
+    marker.color.a = 1.0;
+    marker_array_msg.markers.push_back(marker);
```

Comment:
In order to use it, this node needs to broadcast the object pose as TF.
I want to reduce the burden on /tf topic.